guardian/editions
# Add sharing button

## Summary

We want [the ability to share articles](https://trello.com/c/8F2nykbN/76-social-sharing-share-article-to-social-platforms-and-email). This adds a button if a web URL is available.

Note that I added a sensitive zone around the actual visible button so that it's easier to press it. It's fairly arbitrary and I didn't actually check on a real device, but it's fairly standard practice and reflects what we already do in `button.tsx` with a `hitSlop` of 10px. We can't use our Button component because we're in HTML here, not React Native.

For communicating the share intent, I used `sendMessage`. Another technique would be listen for events on the specific DOM node using an ID, I guess. But in my knowledge it wouldn't be particularly more robust nor simpler.

## Test Plan

1. Open article, check button shows up for most article.
2. Check that the color of the button matches the color swatch of the article.
3. Press and verify sharing works.
4. Verify that it shows up correctly on "large byline" articles (ex. Opinion articles)
5. Repeat step 1-4 on both Android and iOS

<div>
<img width="200" alt="Screenshot 2019-10-30 at 11 50 27" src="https://user-images.githubusercontent.com/1733570/67856653-f2036980-fb0c-11e9-9cea-5215f7db65d2.png">
<img width="200" alt="Screenshot 2019-10-30 at 11 57 23" src="https://user-images.githubusercontent.com/1733570/67856651-f2036980-fb0c-11e9-87f6-804c80bb31fd.png">
<img width="200" alt="Screenshot 2019-10-30 at 12 03 11" src="https://user-images.githubusercontent.com/1733570/67856809-473f7b00-fb0d-11e9-9318-f10bef788581.png">


</div>
